### PR TITLE
Fix mypy tests/providers/amazon/aws/operators

### PR DESCRIPTION
**Pull request by subkanthi**
Related: #19891 

Fixes the following mypy errors in tests/providers/amazon/aws/operators

```
tests/providers/amazon/aws/operators/test_eks.py:95: error: Argument 3 to "EksCreateClusterOperator" has incompatible type "**Dict[str, str]"; expected
"Optional[List[Any]]"
                **self.base_nodegroup_params,
                  ^
tests/providers/amazon/aws/operators/test_eks.py:116: error: Argument 3 to "EksCreateClusterOperator" has incompatible type "**Dict[str, Sequence[Any]]";
expected "Optional[str]"
                **self.base_fargate_profile_params,
                  ^
tests/providers/amazon/aws/operators/test_eks.py:116: error: Argument 3 to "EksCreateClusterOperator" has incompatible type "**Dict[str, Sequence[Any]]";
expected "Optional[List[Any]]"
                **self.base_fargate_profile_params,
                  ^
tests/providers/amazon/aws/operators/test_eks.py:116: error: Argument 3 to "EksCreateClusterOperator" has incompatible type "**Dict[str, Sequence[Any]]";
expected "str"
                **self.base_fargate_profile_params,
                  ^
tests/providers/amazon/aws/operators/test_eks.py:167: error: Argument 2 to "EksCreateFargateProfileOperator" has incompatible type
"**Dict[str, Sequence[Any]]"; expected "str"
                task_id=TASK_ID, **self.create_fargate_profile_params
                                   ^
tests/providers/amazon/aws/operators/test_eks.py:167: error: Argument 2 to "EksCreateFargateProfileOperator" has incompatible type
"**Dict[str, Sequence[Any]]"; expected "List[Any]"
                task_id=TASK_ID, **self.create_fargate_profile_params
                                   ^
tests/providers/amazon/aws/operators/test_eks.py:167: error: Argument 2 to "EksCreateFargateProfileOperator" has incompatible type
"**Dict[str, Sequence[Any]]"; expected "Optional[str]"
                task_id=TASK_ID, **self.create_fargate_profile_params
                                   ^
tests/providers/amazon/aws/operators/test_eks.py:189: error: Argument 2 to "EksCreateNodegroupOperator" has incompatible type "**Dict[str, Sequence[str]]";
expected "str"
                task_id=TASK_ID, **self.create_nodegroup_params
                                   ^
tests/providers/amazon/aws/operators/test_eks.py:189: error: Argument 2 to "EksCreateNodegroupOperator" has incompatible type "**Dict[str, Sequence[str]]";
expected "List[str]"
                task_id=TASK_ID, **self.create_nodegroup_params
                                   ^
tests/providers/amazon/aws/operators/test_eks.py:189: error: Argument 2 to "EksCreateNodegroupOperator" has incompatible type "**Dict[str, Sequence[str]]";
expected "Optional[str]"
                task_id=TASK_ID, **self.create_nodegroup_params
```

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: tests/providers/amazon/aws/operators/test_eks.py [hunk 1]
```diff
@@ -49,10 +52,16 @@
...
+    cluster_name: str
+    cluster_role_arn: str
+    resources_vpc_config: Dict[Any, Any]
+
+
 class TestEksCreateClusterOperator(unittest.TestCase):
     def setUp(self) -> None:
         # Parameters which are needed to create a cluster.
-        self.create_cluster_params = dict(
+        self.create_cluster_params: clusterParams = dict(
```
Comment (subkanthi) [inline tests/providers/amazon/aws/operators/test_eks.py]: @potiuk please let me know your thoughts on this approach for fixing dict unpacking.

Comment (subkanthi) [reply tests/providers/amazon/aws/operators/test_eks.py]: Fixed all errors.

File: tests/providers/amazon/aws/operators/test_eks.py [hunk 2]
```diff
@@ -16,8 +16,11 @@
 # specific language governing permissions and limitations
 # under the License.
 import unittest
+from typing import Any, Dict, List, Optional
 from unittest import mock
 
+from typing_extensions import TypedDict
```
Comment (uranusjr) [inline tests/providers/amazon/aws/operators/test_eks.py]:
```suggestion
from airflow.typing_compat import TypedDict
```

File: tests/providers/amazon/aws/operators/test_eks.py [hunk 3]
```diff
@@ -49,10 +52,41 @@
 NAME_LIST = ["foo", "bar", "baz", "qux"]
 
 
+class clusterParams(TypedDict):
```
Comment (uranusjr) [inline tests/providers/amazon/aws/operators/test_eks.py]:
```suggestion
class ClusterParams(TypedDict):
```

Conventionally types should use CamelCases. (Same for all others below.)

Comment (subkanthi) [reply tests/providers/amazon/aws/operators/test_eks.py]: Fixed, thanks

File: tests/providers/amazon/aws/operators/test_eks.py (hunk 2 above)
Comment (subkanthi) [reply tests/providers/amazon/aws/operators/test_eks.py]: changed, thanks

Comment (subkanthi) [reply tests/providers/amazon/aws/operators/test_eks.py]: Actually @uranusjr , looks like changing this to airflow.typing_compat seems to change the behavior, it introduced these errors, will spend more time testing on both 3.6 and 3.8

`tests/providers/amazon/aws/operators/test_eks.py:88: error: Incompatible types in assignment (expression has type "Dict[str, Collection[Any]]", variable
has type "ClusterParams")
            self.create_cluster_params: ClusterParams = dict(
                                                        ^
tests/providers/amazon/aws/operators/test_eks.py:104: error: Incompatible types in assignment (expression has type "Dict[str, str]", variable has type
"NodeGroupParams")
            self.base_nodegroup_params: NodeGroupParams = dict(
                                                          ^
tests/providers/amazon/aws/operators/test_eks.py:125: error: Incompatible types in assignment (expression has type "Dict[str, Sequence[Any]]", variable has
type "BaseFargateProfileParams")
            self.base_fargate_profile_params: BaseFargateProfileParams = dict(
                                                                         ^
tests/providers/amazon/aws/operators/test_eks.py:183: error: Incompatible types in assignment (expression has type "Dict[str, Sequence[Any]]", variable has
type "CreateFargateProfileParams")
            self.create_fargate_profile_params: CreateFargateProfileParams = dict(
                                                                             ^
tests/providers/amazon/aws/operators/test_eks.py:205: error: Incompatible types in assignment (expression has type "Dict[str, Sequence[str]]", variable has
type "CreateNodegroupParams")
`

**Review by uranusjr — APPROVED** [review]

**Pull request merged**